.claude/hooks/pre_tool_use_protect_precommit.py

```
@@ -0,0 +1,189 @@
+
+def check_content_for_config_changes(content):
+    """Check if file content contains code that would modify configurations."""
+    if not content:
+        return True, None
+    for pattern, reason in DANGEROUS_CONFIG_PATTERNS:
+        if re.search(pattern, content, re.IGNORECASE | re.MULTILINE):
+            return False, reason
+    return True, None
+
+
+def main() -> None:
```

Ruff (0.12.2)

`main` is too complex (11 > 10)

(C901)